gordo_components/cli/client.py

```diff
@@ -38,11 +40,23 @@
     help="Key-Value pair to be entered as metadata labels, may use this option multiple times. "
     "to be separated by a comma. ie: --metadata key,val --metadata some key,some value",
 )
+@click.option(
+    "--session-config",
+    type=yaml.safe_load,
+    default="{}",
+    help="Config json/yaml to set on the requests.Session object. Useful when needing to supply"
+    "authentication parameters such as header keys. ie. --session-config {'headers': {'API-KEY': 'foo-bar'}}",
+)
 @click.pass_context
 def client(ctx: click.Context, *args, **kwargs):
     """
     Entry sub-command for client related activities
     """
+    # Setup the session with any attributes set by the user
+    session = Session()
```

Comment:
Should this maybe just happen if session-config is actually something? I.e. that it is not in general the CLI's task to start the session, only if we actually need it?